File: src/main/kotlin/practice/lec15/Lec15Main.java
package practice.lec15;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Lec15Main {
    public static void main(String[] args){
        int[] array = {100, 200};

        for(int i=0; i<array.length; i++) {
            System.out.printf("%s %s", i, array[i]);
        }
        //---------------------------------------
        final List<Integer> numbers = Arrays.asList(100, 200);

        //하나 가져오기
        System.out.println(numbers.get(0));

        //For Each
        for(int number: numbers) {
            System.out.println(number);
        }

        //전통적인 For문
        for(int i=0; i<numbers.size(); i++) {
            System.out.printf("%s %s", i, numbers.get(i));
        }

//
//        //JDK 8까지
//        Map<Integer, String> oldMap = new HashMap<>();
//        oldMap.put(1, "MONDAY");
//        oldMap.put(2, "TUESDAY");
//
//        //JDK 9부터
//        MAP.of(1, "MONDAY", 2, "TUESDAY");
    }

}

File: src/main/kotlin/practice/lec15/Lec15Main.kt
package practice.lec15

fun main() {
//    val array= arrayOf(100, 200)
//
//    for(i in array.indices) {
//        println("${i} ${array[i]}")
//    }
//
//    for((idx, value) in array.withIndex()) {
//        println("${idx} ${array[idx]}")
//    }

    val numbers = listOf(100, 200)
//    printNumbers(emptyList<Int>())

    //하나 가져오기
    println(numbers[0])

    //For Each
    for(number in numbers){
        println(number)
    }

    for((idx, value) in numbers.withIndex()){
        println("${idx} ${value}")
    }
    //------------------------------------

    val setNumbers = setOf(100, 200)

    //For Each
    for(number in setNumbers){
        println(number)
    }
    //전통적인 for문
    for((index, number) in setNumbers.withIndex()){
        println("${index} ${number}")
    }

    //---------------------------------------------
    val oldMap = mutableMapOf<Int, String>()
    oldMap[1]="MONDAY"
    oldMap[2]="TUESDAY"

    mapOf(1 to "MONDAY", 2 to "TUESDAY")
    //-----------------------------------
    for(key in oldMap.keys) {
        println(key)
        println(oldMap[key])
    }

    for((key, value) in oldMap.entries) {
        println(key)
        println(value)
    }
}

//private fun printNumbers(numbers : List<Int>){}